src/middleware/arns.ts

```diff
@@ -25,33 +25,42 @@ const EXCLUDED_SUBDOMAINS = new Set('www');
 
 export const createArnsMiddleware = ({
   dataHandler,
+  rootHost,
   nameResolver,
 }: {
   dataHandler: Handler;
+  rootHost: string;
   nameResolver: NameResolver;
 }): Handler =>
   asyncMiddleware(async (req, res, next) => {
+    const rootHostSubdomainLength = rootHost.split('.').length - 2;
```

Comment:
@kay-is, let's reduce duplication by calculating this value in `config.ts` (create a  `ROOT_HOST_SUBDOMAIN_LENGTH` const) and then passing that into the middlewares. Alternately, you could create a utility function and use it in both locations, but I think the config constant approach is probably simpler for now.

Reply:
I thought about doing this, but I didn't want to contaminate app.js with implementation details of the middleware. If you okay with that, I'll change it.